runtime/lua/vim/lsp/util.lua

```diff
@@ -2229,6 +2230,22 @@ function M._refresh(method, opts)
   end
 end
 
+---@private
+--- Get clients attached to a buffer that support a given method.
+---
+---@param bufnr integer buffer number
+---@param method string LSP method
+function M._get_clients_with_method(bufnr, method)
```

Comment:
get_clients supports a `method` filter that can be used instead of this new method